pkg/gateway/operations/operation_utils.go

```diff
@@ -22,13 +26,21 @@ var (
 func amzMetaAsMetadata(req *http.Request) (catalog.Metadata, error) {
 	metadata := make(catalog.Metadata)
 	var err error
+
+	var userMetadataSize int
 	for k := range req.Header {
-		if strings.HasPrefix(k, amzMetaHeaderPrefix) {
+		name, found := strings.CutPrefix(k, amzMetaHeaderPrefix)
+		if found {
 			value, decodeErr := rfc2047Decoder.DecodeHeader(req.Header.Get(k))
 			if decodeErr != nil {
 				err = errors.Join(err, decodeErr)
 				continue
 			}
+			userMetadataSize += len(name) + len(value)
+			if userMetadataSize > maxUserMetadataSize {
+				err = gatewayErrors.ErrMetadataTooLarge
```

Comment:
I was not fully sure about the error conventions here. Since this is a utility function, I didn’t create a new error type. 
I'm happy to hear suggestions on what would be best here.